GitUI/GitUICommands.cs

```diff
@@ -1675,25 +1637,27 @@ private void RunOpenRepoCommand(IReadOnlyList<string> args)
                 }
             }
 
-            c.StartBrowseDialog(null, GetParameterOrEmptyStringAsDefault(args, "-filter"));
+            return c.StartBrowseDialog(null, GetParameterOrEmptyStringAsDefault(args, "-filter"));
         }
 
-        private void RunSynchronizeCommand(IReadOnlyDictionary<string, string> arguments)
+        private bool RunSynchronizeCommand(IReadOnlyDictionary<string, string> arguments)
         {
-            Commit(arguments);
-            Pull(arguments);
-            Push(arguments);
+            bool successful = true;
+            successful = Commit(arguments) && successful;
+            successful = Pull(arguments) && successful;
+            successful = Push(arguments) && successful;
```

Comment:
I'm not sure this is correct. Suppose `Commit` will fail, so the `successful` on line:1646 will be `false`.
We will still execute `Pull`, which would be incorrect, wouldn't it?

Should it be this?
```suggestion
            bool successful = Commit(arguments) && Pull(arguments) && Push(arguments);
```

Reply:
Stopping this sequence might make more sense.
Though I do not want to change the behavior in this commit, but only return the result.
Before, all three commands were run unconditionally one after the other:
```
            Commit(arguments);
            Pull(arguments);
            Push(arguments);
```

Reply:
Should we fix it now?

Reply:
Do it in a separate issue/PR
Changing this may break some specific usage and it is simple to separate it